sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -142,15 +142,16 @@ static az_result faulted(az_event_policy* me, az_event event)
 
 AZ_INLINE az_result _start_connect(az_mqtt5_connection* me)
 {
-  az_mqtt5_x509_client_certificate* current_client_certificate
-      = &me->_internal.options.client_certificates
-             [me->_internal.client_certificate_index
-              % me->_internal.options.client_certificate_count];
+
+  az_mqtt5_x509_client_certificate* current_client_certificate = NULL;
```

Comment:
I don't think you need to set this outside of the if statement, since it's only used inside of the if statement!

Reply:
Set outside! Thanks for catching that